modmesh/onedim/euler1d.py

```diff
@@ -302,16 +308,22 @@ def calc_internal_energy(self, pressure, density):
     def calc_entropy(self, pressure, density):
         return pressure / (density ** self.gamma)
 
-    def build_field(self, t, coord=None):
+    def build_field(self, t, coord=None, keep_edge=False):
         """
         Populate the field data using the analytical solution.
 
         :param t:
         :param coord: If None, take the coordinate from the numerical solver.
         :return: None
         """
-        if None is coord:
-            coord = self.svr.coord[::2]  # Use the numerical solver.
+        if coord is None:
+            _ = self.svr.ncoord - 1
```

Comment:
Why do you need to repeat the same code here and in [line 32](https://github.com/solvcon/modmesh/pull/391/files#diff-00396195422e4a46224688f7c4b3aa59a6ceedd1d4c37cc274d4e86597737561R32)?